contracts/test/StakeTable.t.sol

```diff
@@ -530,4 +553,162 @@ contract StakeTable_Test is Test {
+                seedUsed = true;
+                break;
+            }
+        }
+
+        if (seedUsed) {
+            registerWithSeed(sender, rands[i], randDepositAmount, true);
+            return false;
+        } else {
+            (BN254.G2Point memory blsVK,) =
+                registerWithSeed(sender, rands[i], randDepositAmount, false);
+            registeredKeys[i] = blsVK;
+            isKeyActive[i] = true;
+            // Invariants specific to a successful registration
+            assertGe(stakeTable.firstAvailableRegistrationEpoch(), stakeTable.currentEpoch() + 1);
+            assertGe(stakeTable.numPendingRegistrations(), 1);
+            return true;
+        }
+    }
+
+    ///@dev Test invariants about our queue logic holds during a random sequence of register,
+    /// requestExit, and advanceEpoch operations
+    function testFuzz_SequencesOfEvents(
+        uint8[ARRAY_SIZE] memory events,
+        uint8[ARRAY_SIZE] memory rands
+    ) external {
+        BN254.G2Point[ARRAY_SIZE] memory registeredKeys;
+
+        // Tracks the indices corresponding to an active key
+        bool[ARRAY_SIZE] memory isKeyActive;
+
+        uint64 numRegistrations = 0;
+        uint64 numExits = 0;
+
+        for (uint256 i = 0; i < ARRAY_SIZE; i++) {
+            uint256 ev = bound(events[i], 0, 2);
+
+            bool exitRequestSuccessful = false;
+
+            if (ev == 0) {
+                // Registrations
+                bool res = handleRegistrations(i, rands, registeredKeys, isKeyActive);
+                if (res) {
+                    numRegistrations++;
+                }
+            } else if (ev == 1) {
+                // Exits
+                if (numRegistrations == 0) {
+                    break;
+                }
+                uint256 indexRegistration = bound(rands[i], 0, numRegistrations - 1);
+                bytes32 hashNode = stakeTable._hashBlsKey(registeredKeys[indexRegistration]);
+                (
+                    address sender,
+                    AbstractStakeTable.StakeType stakeType,
+                    uint64 balance,
+                    uint64 registerEpoch,
+                    uint64 exitEpoch,
+                ) = stakeTable.nodes(hashNode);
+
+                balance;
+                stakeType;
+
+                BN254.G2Point memory blsVK = registeredKeys[indexRegistration];
+
+                bool canExit = (stakeTable.currentEpoch() >= registerEpoch + 1) && (exitEpoch == 0);
+                if (canExit) {
+                    vm.prank(sender);
+                    bool res = stakeTable.requestExit(blsVK);
+                    assertTrue(res);
+                    numExits++;
+                    exitRequestSuccessful = true;
+
+                    // Invariants specific to a successful exit
+                    assertGe(stakeTable.firstAvailableExitEpoch(), stakeTable.currentEpoch() + 1);
+                    assertGe(stakeTable.numPendingExits(), 1);
```

Comment:
I think this invariant should be, we first record down before exit , this exit queue size, and then now, make sure this queue size increment by 1

Reply:
This is not always the case. For example we may have the queue size equal to 4 during epoch 10. Then nothing happens during epochs 11,12,13. The next queue size will be 1 instead of 5 (assuming `max_churn_rate == 10` for example).

Reply:
As mentioned [above](https://github.com/EspressoSystems/espresso-sequencer/pull/803#discussion_r1427899546), with the `skipEpoch` fuzzing variable we can implement your idea.